### ISSUE TITLE
Anonlink support on MacOS

### ISSUE THREAD

**Issue by hardbyte** [data61/anonlink#247]
CI on MacOS is currently broken so we no longer test and publish MacOS wheels.

This would be good to fix

**Comment by hardbyte** [comment]
Currently the issue (using the fork of cibuildwheel mentioned in #242) seems to be unrolling the inline assembly demands too many registers:

```
  building '_entitymatcher' extension
  creating build/temp.macosx-10.6-intel-3.6/build
  creating build/temp.macosx-10.6-intel-3.6/build/temp.macosx-10.6-intel-3.6
  gcc -fno-strict-aliasing -Wsign-compare -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch i386 -arch x86_64 -g -I_cffi_build -I/Library/Frameworks/Python.framework/Versions/3.6/include/python3.6m -c build/temp.macosx-10.6-intel-3.6/_entitymatcher.cpp -o build/temp.macosx-10.6-intel-3.6/build/temp.macosx-10.6-intel-3.6/_entitymatcher.o -Wall -Wextra -Werror -O3 -std=c++11 -fvisibility=hidden
  build/temp.macosx-10.6-intel-3.6/_entitymatcher.cpp:553:9: error: inline assembly requires more registers than available
          "popcnt %4, %4  \n\t"
          ^
  build/temp.macosx-10.6-intel-3.6/_entitymatcher.cpp:553:9: error: inline assembly requires more registers than available
```

**Comment by hardbyte** [comment]
The issue is cibuildwheel actually tries to build both 32 and 64bit versions  for MacOS and as far as I can tell isn't configurable (yet). See upstream issue https://github.com/joerick/cibuildwheel/pull/220 in particular this [comment](https://github.com/joerick/cibuildwheel/pull/220#discussion_r361488010) regarding the changing build identifiers.

In branch https://github.com/data61/anonlink/tree/update-cibuildwheel I've enabled MacOS Python3.8 wheel builds and tests.

**Comment by wilko77** [comment]
addressed in #302
